docs/reference/eql/eql-search-api.asciidoc

```diff
@@ -56,9 +56,13 @@ To search all data streams and indices in a cluster, use
 [[eql-search-api-query-params]]
 ==== {api-query-parms-title}
 
-include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
-+
-Defaults to `false`.
+`allow_no_indices`::
+(Optional, boolean)
+If `false`, the request returns a `mapping_exception` error when a wildcard
+expression, <<indices-aliases,index alias>>, or `_all` value targets only
+missing or closed indices. If `true`, the request returns a
+`verification_exception` error when targeting only missing or closed indices.
+Defaults to `true`.
```

Comment:
@astefan Let me know if I missed anything here. I'm not sure there is much value in users changing this arg as there will be an error either way.

Reply:
It is correct what you have here, but incomplete. The situation is a bit more complicated and it matters if the cluster is secured or not. With Security enabled and `allow_no_indices=false` for a `inexistent*,existent` search there is a `index_not_found_exception` and this one comes from Security itself if I'm not mistaken.
What we have in eql atm is a variation of `allow_no_indices` behavior in ES itself. We do obey the default `allow_no_indices` behavior BUT in case the entire pattern doesn't have any index, no matter the value of `allow_no_indices`, EQL complains. Elasticsearch, on the other hand, has scenarios where it returns an empty response whereas we return a `verification_exception`.

Reply:
Thanks for the clarification @astefan. After doing some testing, I'm not sure that having security enabled is a big differentiator. I also don't think I need to call out the individual exception names. Instead, I focused on scenarios that would generate an error vs. those that would not.